pkg/token/token.go

```diff
@@ -231,13 +248,15 @@ func (g generator) GetWithSTS(clusterID string, stsAPI *sts.STS) (string, error)
 
 // FormatJSON formats the json to support ExecCredential authentication
 func (g generator) FormatJSON(token string) string {
+	expirationTimestamp := metav1.NewTime(time.Now().Local().Add(refreshTokenAfter))
 	execInput := &clientauthv1alpha1.ExecCredential{
 		TypeMeta: metav1.TypeMeta{
 			APIVersion: "client.authentication.k8s.io/v1alpha1",
 			Kind:       "ExecCredential",
 		},
 		Status: &clientauthv1alpha1.ExecCredentialStatus{
-			Token: token,
+			ExpirationTimestamp: &expirationTimestamp,
```

Comment:
What happens is the FormatJSON is executed each time a token is generated, and the response is cached in client-go.  This will solve the issue where client-go is running continuously, like in the sample program provided in the issue, or in kubelet, etc.  For kubectl, we need client caching in a file, for which there is currently another PR open.  I did refactor a little based on your comment (no functionality change) where its now generating the expiration next to the place where the token is generated.  The expiration of the token is based on the `x-amz-date` header, so I think I'll try deriving the expiration from that instead, which is technically correct, though in practice I doubt there would be a noticeable difference from the two timestamps.